plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -322,35 +383,57 @@ public void shutdown(NetworkProfile profile, NetworkOffering offering) {
+        if (network == null) {
+            throw new ConcurrentOperationException("Unable to acquire lock on network " + networkId);
         }
+
         try {
-            HostVO nuageVspHost = getNuageVspHost(network.getPhysicalNetworkId());
-            TrashNetworkVspCommand cmd = new TrashNetworkVspCommand(domain.getUuid(), network.getUuid(), isL3Network(offering.getId()), vpcUuid);
-            TrashNetworkVspAnswer answer = (TrashNetworkVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmd);
-            if (answer == null || !answer.getResult()) {
-                s_logger.error("TrashNetworkNuageVspCommand for network " + network.getUuid() + " failed");
-                if ((null != answer) && (null != answer.getDetails())) {
-                    s_logger.error(answer.getDetails());
+            s_logger.debug("Handling trash() call back to delete the network " + network.getName() + " with uuid " + network.getUuid() + " from VSP");
+            long domainId = network.getDomainId();
+            Domain domain = _domainDao.findById(domainId);
+            boolean isL3Network = isL3Network(network);
+            boolean isSharedNetwork = offering.getGuestType() == GuestType.Shared;
+            Long vpcId = network.getVpcId();
+            String vpcUuid = null;
+            if (vpcId != null) {
+                Vpc vpcObj = _vpcDao.findById(vpcId);
+                vpcUuid = vpcObj.getUuid();
+            }
+
+            String preConfiguredDomainTemplateName = null;
+            if (vpcUuid != null) {
+                preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspVpcDomainTemplateName.key());
+            } else if (isSharedNetwork) {
+                preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspSharedNetworkDomainTemplateName.key());
+            } else if (isL3Network) {
+                preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspIsolatedNetworkDomainTemplateName.key());
+            }
+
+            try {
+                HostVO nuageVspHost = getNuageVspHost(network.getPhysicalNetworkId());
+                TrashNetworkVspCommand cmd = new TrashNetworkVspCommand(domain.getUuid(), network.getUuid(), isL3Network, isSharedNetwork, vpcUuid, preConfiguredDomainTemplateName);
+                TrashNetworkVspAnswer answer = (TrashNetworkVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmd);
+                if (answer == null || !answer.getResult()) {
+                    s_logger.error("TrashNetworkNuageVspCommand for network " + network.getUuid() + " failed");
+                    if ((null != answer) && (null != answer.getDetails())) {
+                        s_logger.error(answer.getDetails());
+                    }
                 }
+            } catch (Exception e) {
+                s_logger.warn("Failed to clean up network information in Vsp " + e.getMessage());
```

Comment:
Add the exception, ``e``, to the ``warn`` call to provide the stack trace in the log to assist debugging operations.